tests/std/tests/Dev09_126254_persistent_aux_allocators/test.cpp

```diff
@@ -98,6 +98,12 @@ void dump_map() {
     }
 }
 
+void test_gh_2362() {
+    allocator<int> al;
+    int* ptr = al.allocate(0);
+    al.deallocate(ptr, 0);

```

Comment:
I'd also like to see a test for an over-aligned type since they go through some different codepaths.